cron_scripts/password_expiration_notification.pl

```diff
@@ -0,0 +1,290 @@
+    localtime( time() - $notification_compare_days * ( 24 * 60 * 60 ) ) )
+  . "Z";
+
+my $soap_api = ZCS::CustomAPI->new(
+    conf => {
+        SOAPURI => "https://"
+          . $config->{MAILSTOREHOST}
+          . ":7071/service/admin/soap",
+        SOAPUser => $config->{MAILSTOREUSER},
+        SOAPPass => $config->{MAILSTOREPWD},
+    },
+    debug => $Debug
+);
+
+# search for accounts whose status are either of locked, lockout or pending
+# and last login is earlier than specified days $last_login_days
+my @data;
+if ($soap_api) {
+    my %args = (
+        "query" =>
+"(&(zimbraPasswordModifiedTime<=$notification_timestamp)(zimbraAccountStatus=active)(!(zimbraIsAdminAccount=TRUE))(!(zimbraIsDelegatedAdminAccount=TRUE)))",
+        "attrs" =>
+          "uid,zimbraId,mobile,mail,displayName,zimbraPasswordModifiedTime"
+    );
+
+    if ( my $searched_users = eval { $soap_api->searchdirectory(%args) } ) {
+        my $path = "/Envelope/Body/SearchDirectoryResponse";
+        my $i    = 0;
+        foreach my $acc ( $searched_users->dataof("$path/account") ) {
+            $i++;
+            my $msom = $searched_users->match("$path/[$i]")
+              or die("MATCH '$path/[$i]' failed\n");
+
+            my $users_attr = getSoapResponseData($acc);
+            if ( defined $users_attr->{'zimbraPasswordModifiedTime'} ) {
+                my $zimbraPasswordModifiedTime =
+                  $users_attr->{'zimbraPasswordModifiedTime'};
+                $zimbraPasswordModifiedTime =
+                  substr $zimbraPasswordModifiedTime, 0, 12;
+                my $pass_expiration_year_days_users =
+                  Time::Piece->strptime( $zimbraPasswordModifiedTime,
+                    "%Y%m%d%H%M%S" );
+                my $expire_in_days =
+                  $pass_expiration_year_days -
+                  $pass_expiration_year_days_users->yday;
+                $users_attr->{'expire_in_days'} = $expire_in_days;
+            }
+            push @data, $users_attr;
+        }
+    }
+    else {
+        warn(   "Error: SearchDirectoryRequest "
+              . $soap_api->Error
+              . " via $config->{MAILSTOREHOST} \n" );
+    }
+    if ( scalar @data != 0 ) {
+        my $pm = Parallel::ForkManager->new( $opts{max_proc} );
+      ACCOUNTS:
+        foreach my $acc (@data) {
+            $pm->start and next ACCOUNTS;
+            my $mail           = $acc->{mail};
+            my $mobile         = $acc->{mobile};
+            my $displayName    = $acc->{displayName} || 'User';
+            my $expire_in_days = $acc->{'expire_in_days'};
+            if ( $mail ) {
+                my $mail_content = $config->{mail_content};
+                $mail_content =~ s/{displayName}/$displayName/g;
+                $mail_content =~ s/{numDays}/$expire_in_days/g;
+                my $mail_data = {
+                    'to' => $mail,
+                    'subject' =>
+"Your Mail access password expire in $acc->{'expire_in_days'} days",
```

Comment:
nit: grammar needs work.  Perhaps (add 's' to expire, possibly add parenthesis around the 's' at the end of days):
- ```Your Mail access password expires in $acc->{'expire_in_days'} day(s)```

I'm not a huge fan of that still, but whatever the customer is comfortable with should be fine.